tracing/tracer.go

```diff
@@ -100,3 +103,17 @@ func (jr *jaegerReporter) UpdateGauge(name string, tags map[string]string, value
 func (jr *jaegerReporter) RecordTimer(name string, tags map[string]string, d time.Duration) {
 	jr.reporter.AllocateTimer(name, tags).ReportTimer(d)
 }
+
+// WithSpan is used for generating a span to be used in testing
+func WithSpan(t *testing.T, log ulog.Log, f func(opentracing.Span)) {
+	tracer, closer, err := InitGlobalTracer(
+		nil, "serviceName", log, metrics.NopCachedStatsReporter,
+	)
+	assert.NoError(t, err)
+	defer closer.Close()
+	opentracing.InitGlobalTracer(tracer)
+	defer opentracing.InitGlobalTracer(opentracing.NoopTracer{})
+	span := opentracing.GlobalTracer().StartSpan("test")
+	defer span.Finish()
```

Comment:
this is part tracing confuse me. `span.Finish` will report to tracer's reporter through a channel, but tracer is closed first and close the channel as well. However, this will still work somehow...